apis/pkg/v1alpha1/configuration_types.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// +kubebuilder:object:root=true
+// +genclient
+// +genclient:nonNamespaced
+
+// Configuration is the CRD type for a request to add a configuration to Crossplane.
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="PACKAGE",type="string",JSONPath=".spec.package"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:resource:scope=Cluster,categories={crossplane}
+type Configuration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ConfigurationSpec   `json:"spec,omitempty"`
+	Status ConfigurationStatus `json:"status,omitempty"`
+}
+
+// ConfigurationSpec specifies details about a request to install a
+// configuration to Crossplane.
+type ConfigurationSpec struct {
+	// Package is the name of the package that is being requested, e.g.,
+	// myapp. Either Package or CustomResourceDefinition can be specified.
+	Package string `json:"package"`
+
+	// RevisionActivationPolicy specifies how the package controller should
+	// update from one revision to the next. Options are Automatic or Manual.
+	// Default is Automatic.
+	RevisionActivationPolicy *RevisionActivationPolicy `json:"revisionActivationPolicy,omitempty"`
+
+	// RevisionHistoryLimit dictates how the package controller cleans up old
+	// inactive package revisions.
+	// Defaults to 1. Can be disabled by explicitly setting to 0.
+	RevisionHistoryLimit *int64 `json:"revisionHistoryLimit,omitempty"`
+
+	// ImagePullSecrets are named secrets in the same workspace that can be used
+	// to fetch Configurations from private repositories.
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+	// ImagePullPolicy defines the pull policy for all images used during
+	// Configuration extraction.
+	// https://kubernetes.io/docs/concepts/configuration/overview/#container-images
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
```

Comment:
That sounds fine to me. It makes it a bit more abstract for the user, but since we are already intentionally introducing abstraction, I would be fine to continue that trend with these fields.